acl/role.go

```diff
@@ -0,0 +1,8 @@
+package acl
+
+type Role interface {
+	Name() string
+	ACL() ACL
+	// root role has the capacity to manage roles and users
+	Root() bool
```

Comment:
Manage is the capability to grant/revoke an entry in an ACL (which is attached to a role, since we have 1-1 mapping). 

Root can manage roles and users.